### PR TITLE
Handle dropping the active file while Vifm is open

### LINKED ISSUE #96: Files are opened in the wrong window on exit if files are deleted while using Vifm in a terminal
Edit: may actually happen in other modes, not just embedded terminal mode

My `init.lua`:

```
require("pckr").add({
  {
    "vifm/vifm.vim",
    config_pre = function()
      vim.g.vifm_embed_term = 1
      vim.g.vifm_embed_split = 1
      vim.g.vifm_embed_cwd = 1
      vim.g.vifm_replace_netrw = 1
      vim.g.loaded_netrw = 0
      vim.g.loaded_netrwPlugin = 0
    end
  },
})
```

Steps to reproduce:

* Run `:e FILE_A` in Vim
* Make changes and save
* Run `:e FILE_B` in Vim
* Make changes and save
* Run `:Vifm`
* Delete `FILE_B`
* Open `FILE_C`

Expected result:

* `FILE_C` is opened in the window that `:Vifm` was opened on (`FILE_B`)
* The window that `FILE_A` was opened in is not touched because `:Vifm` was not run from this window

Actual result:

* `FILE_C` is opened in the window that `FILE_A` was open in
* The window containing `FILE_B` is closed

This is caused by fixes introduced for #85 which wipes out any buffers deleted while Vifm is open.

### PR DESCRIPTION
**Pull request by rbong**
Fixes #96

### CODE REVIEW

File: plugin/vifm.vim [hunk 1]
```diff
@@ -255,6 +255,74 @@ function s:DropGoneBuffers(buffer_snapshot) abort
 	endif
 endfunction
 
+" Opens files after exiting Vifm.  Returns a dict of opened buffer names.
+function! s:OpenFiles(editcmd, flist, opentype) abort
```
Comment (rbong) [inline plugin/vifm.vim:259]: I had to extract the logic in `HandleRunResults` top open files into its own function. This is to make flow control a little cleaner in `HandleRunResults`.

File: plugin/vifm.vim [hunk 2]
```diff
@@ -255,6 +255,74 @@ function s:DropGoneBuffers(buffer_snapshot) abort
...
+	let opened_bufnrs = {}
+
+	" User exits vifm without selecting a file.
+	if empty(a:flist)
+		echohl WarningMsg | echo 'No file selected' | echohl None
+		return opened_bufnrs
+	endif
+
+	let flist = a:flist
+	call map(flist, 'resolve(fnamemodify(v:val, ":."))')
```
Comment (rbong) [inline plugin/vifm.vim:269]: I changed some of the logic for handling escaping and modifying file names in this function.

Previously, `resolve()` and `fnamemodify()` were called after `fnameescape()`. These functions don't actually work with escaped filenames.

The `resolve()` function wasn't actually called in one case, and there was special logic for resolving filenames in one case. However, I discovered this was still broken in my tests - this logic broke with filenames that needed to be escaped or linked files (it only worked with files in linked folders that didn't need to be escaped).

Now, `resolve()` and `fnameescape()` are used properly on all files, and `fnameescape()` is now called when it's needed.

File: doc/plugin/vifm-plugin.txt [hunk 3]
```diff
@@ -108,6 +108,10 @@ opening directories using vifm.  Defaults to false.
 A string variable.  The command to use to open folders when
 g:vifm_replace_netrw is enabled.  Defaults to "Vifm".
 
+                                               *g:vifm_drop_gone_buffers*
+A boolean variable.  When true, upon exiting Vifm, previously open buffers
```
Comment (xaizek) [inline doc/plugin/vifm-plugin.txt:112]: This needs to say that only unmodified buffers get deleted, makes a difference.

File: plugin/vifm.vim (hunk 2 above)
Comment (xaizek) [reply plugin/vifm.vim:269]: Thank you for cleaning up that mess!  I'm not sure why I got the impression that Vim was resolving only part of the path there.

File: doc/plugin/vifm-plugin.txt (hunk 3 above)
Comment (rbong) [reply doc/plugin/vifm-plugin.txt]: Thanks, should be fixed now.

**Comment by xaizek** [comment]
Thanks!

**Pull request merged**